Trip.py

```diff
@@ -30,15 +32,39 @@ def _build_legs(self):
             else:
                 self._legs[i] = Leg(qp, [segment['travel_mode']])
 
+    def _find_best_mode(self):
+        for mode in self._acceptable_modes:
+            qp = QueryParser(self._query_parser.get_start(),
```

Comment:
So we define a query parser for the original trip and then subsequent QueryParsers are used to build the best mode/route? Only asking to follow.